lib/rules/jsx-no-leaked-zero.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @fileoverview Prevent zero from being rendered on numerical condition
+ * @author Mario Beltrán
+ */
+
+'use strict';
+
+const docsUrl = require('../util/docsUrl');
+const report = require('../util/report');
+const isParenthesised = require('../util/ast').isParenthesised;
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const messages = {
+  noPotentialNumericEvaluation: 'Potential numeric evaluation that ends up in leaked zero to the DOM',
```

Comment:
react isn't just about the DOM; this also applies to server rendering.
```suggestion
  noPotentialNumericEvaluation: 'Potential numeric evaluation resulting in an unintentionally rendered `0`',
```